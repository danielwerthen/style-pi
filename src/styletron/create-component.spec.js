const React = require('react');
const ReactDOMServer = require('react-dom/server');
const createComponent = require('./create-component');
const Style = require('./style');
const Provider = require('./provider');
const Styletron = require('styletron-engine-atomic').Server;

describe('CreateComponent', () => {
  describe('Simple scenario', () => {
    class MyStyle extends Style {}

    it('should work', () => {
      const Component = createComponent({
        Style: MyStyle,
      })('div');
      const styletron = new Styletron();
      const result = ReactDOMServer.renderToString(
        React.createElement(Provider, {
          value: styletron,
          children: React.createElement(Component, {
            children: 'Daniel',
            fontSize: '14px',
            fontFamily: 'Comic Sans',
          }),
        }),
      );

      expect(result).toMatchSnapshot();
      expect(styletron.getCss()).toMatchSnapshot();
    });
  });
});
