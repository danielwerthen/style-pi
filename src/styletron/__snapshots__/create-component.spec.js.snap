// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateComponent Simple scenario should work 1`] = `"<div class=\\"ae ae\\" data-reactroot=\\"\\">Daniel</div>"`;

exports[`CreateComponent Simple scenario should work 2`] = `".ae{undefined:undefined}"`;
